include/writer/kernel.tpp

```diff
@@ -126,36 +125,35 @@ void specfem::writer::kernel<OutputLibrary>::write() {
     DomainView c35("c35", n_elastic_anisotropic, ngllz, ngllx);
     DomainView c55("c55", n_elastic_anisotropic, ngllz, ngllx);
 
-    int i = 0;
-
-    for (int ispec = 0; ispec < nspec; ispec++) {
-      if (kernels.h_element_types(ispec) ==
-          specfem::element::medium_tag::elastic &&
-          kernels.h_element_property(ispec) == specfem::element::property_tag::anisotropic) {
-        for (int iz = 0; iz < ngllz; iz++) {
-          for (int ix = 0; ix < ngllx; ix++) {
-            x(i, iz, ix) = mesh.points.h_coord(0, ispec, iz, ix);
-            z(i, iz, ix) = mesh.points.h_coord(1, ispec, iz, ix);
-            const specfem::point::index<specfem::dimension::type::dim2> index(
-                ispec, iz, ix);
-            specfem::point::kernels<specfem::dimension::type::dim2,
-                                    specfem::element::medium_tag::elastic,
-                                    specfem::element::property_tag::anisotropic,
-                                    false>
-                point_kernels;
-
-            specfem::compute::load_on_host(index, kernels, point_kernels);
-
-            rho(i, iz, ix) = point_kernels.rho;
-            c11(i, iz, ix) = point_kernels.c11;
-            c13(i, iz, ix) = point_kernels.c13;
-            c15(i, iz, ix) = point_kernels.c15;
-            c33(i, iz, ix) = point_kernels.c33;
-            c35(i, iz, ix) = point_kernels.c35;
-            c55(i, iz, ix) = point_kernels.c55;
-          }
+    const auto element_indices = kernels.get_elements_on_host(
+        specfem::element::medium_tag::elastic,
+        specfem::element::property_tag::anisotropic);
+    const int nelements = element_indices.size();
+
+    for (int i = 0; i < nelements; i++) {
```

Comment:
This is just reducing the loop to be over just the anisotropic elements, instead of all elements checking which ones are anisotropic, correct?

Reply:
The loop for getting the elements is the same, it is moved to a function because it can be reused.